In [35]:
import numpy as np
import pandas as pd
df = pd.read_csv('npadata.csv')
from IPython.core.display import display, HTML
display(HTML("<style>.container { width:100% !important; }</style>"))

#df1 = just country, requirements, duration
df1 = df[['country', 'requirements', 'degree', 'duration']]

# First, let's have a look at the data
Here we have it sorted by country

In [3]:
df.sort_values('country')

Source  Order                   country  \
94   redcap2018    127                    Angola   
93   redcap2018    113                    Angola   
33   redcap2018    106                    Angola   
95   redcap2018    111                     Benin   
85          SSA      1                     Benin   
87   redcap2018     86                     Benin   
86   redcap2018     87                     Benin   
34         AAAC     79                     Benin   
96   redcap2018    112                     Benin   
35          SSA      2                  Botswana   
36         AAAC     72              Burkina Faso   
97   redcap2018    118              Burkina Faso   
38          SSA      5              Burkina Faso   
39   redcap2018     93              Burkina Faso   
37          SSA      4              Burkina Faso   
40          SSA      3                   Burundi   
3           SSA      6                   Burundi   
80          SSA      7                   Burundi   
41         AAAC     76                  Cameroon   
42          SSA      8                  Cameroon   
88   redcap2018    100  Central African Republic   
110         SSA     10                      Chad   
1    redcap2018     98       Congo (Brazzaville)   
44   redcap2018     97             Cote D'Ivoire   
43          SSA      9             Cote D'Ivoire   
98   redcap2018    120             Cote D'Ivoire   
99   redcap2018    123                       DRC   
2           SSA     11                       DRC   
100  redcap2018    124                       DRC   
45   redcap2018    102                       DRC   
..          ...    ...                       ...   
63         AAAC     75              South Africa   
124         SSA     49              South Africa   
64          SSA     51               South Sudan   
20         AAAC     71                     Sudan   
21          SSA     52                 Swaziland   
28          SSA     55                  Tanzania   
29         AAAC     81                  Tanzania   
66          SSA     53                  Tanzania   
65         AAAC     82                  Tanzania   
30          SSA     54                  Tanzania   
78          SSA     56                      Togo   
91   redcap2018     94                      Togo   
0    redcap2018     92                    Uganda   
108  redcap2018    114                    Uganda   
22   redcap2018    108                    Uganda   
31          SSA     60                    Uganda   
67          SSA     58                    Uganda   
68          SSA     62                    Uganda   
69   redcap2018     88                    Uganda   
109  redcap2018    116                    Uganda   
70          SSA     59                    Uganda   
79          SSA     57                    Uganda   
126         SSA     61                    Uganda   
72         AAAC     77                    Zambia   
73          SSA     64                    Zambia   
71          SSA     63                    Zambia   
74          SSA     66                  Zimbabwe   
75          SSA     67                  Zimbabwe   
32          SSA     65                  Zimbabwe   
76         AAAC     73                  Zimbabwe   

                                          requirements  duration  \
94                                      Medical school      48.0   
93                                      Medical school       NaN   
33                                     Nursing diploma      36.0   
95                                      Medical school       NaN   
85                                     Nursing diploma      36.0   
87                                     Nursing diploma      24.0   
86                                     Nursing diploma      24.0   
34                                             Diploma      72.0   
96                                      Medical school       NaN   
35                                      Nursing degree      12.0   
36                                  

It appears there are 127 possible separate programs

# How many have blanks in the core categories of requirements, duration, or degree?

These are the entries with something missing:

In [80]:
missing = df1[(df.duration.isnull()==True) | (df['requirements'].isnull()==True) | (df['degree'].isnull()==True)].sort_values('country')
missing

country         requirements                degree  duration
93         Angola       Medical school  physician specialist       NaN
95          Benin       Medical school  physician specialist       NaN
96          Benin       Medical school  physician specialist       NaN
80        Burundi                  NaN               Licence       NaN
110          Chad                  NaN                   NaN       NaN
8        Ethiopia                  NaN     Bachelor's degree       NaN
9        Ethiopia                  NaN     Bachelor's degree       NaN
102      Ethiopia       Medical school  physician specialist       NaN
10         Gambia      Nursing diploma     Bachelor's degree       NaN
111         Ghana                  NaN                   NaN       NaN
114         Kenya                  NaN                   NaN       NaN
113         Kenya                  NaN                   NaN       NaN
112         Kenya                  NaN                   NaN       NaN
115        Malawi                  NaN                   NaN       NaN
116     Mauritius                  NaN                   NaN       NaN
117       Namibia                  NaN                   NaN       NaN
118         Niger                  NaN                   NaN       NaN
119        Rwanda                  NaN                   NaN       NaN
120        Rwanda                  NaN                   NaN       NaN
121  Sierra Leone                  NaN                   NaN       NaN
122       Somalia      Nursing diploma                   NaN      18.0
123       Somalia                  NaN                   NaN       NaN
125  South Africa                  NaN                   NaN       NaN
124  South Africa                  NaN                   NaN       NaN
79         Uganda                  NaN          I don't know      12.0
126        Uganda                  NaN                   NaN       NaN
72         Zambia  RN, medical diploma               Diploma       NaN
32       Zimbabwe                  NaN           Certificate       NaN

In [81]:
missingnum = missing['country'].count()
print('There are ', missingnum, 'rows with missing core elements')

There are  28 rows with missing core elements


# Therefore, there are 28 entries with missing elements
# Let's drop them from the database, the new database is df2

In [86]:
df2 = df1[(df.duration.isnull()==False) & (df.requirements.isnull()==False) & (df.degree.isnull()==False)].sort_values('country')
df2

country  \
33                     Angola   
94                     Angola   
34                      Benin   
87                      Benin   
86                      Benin   
85                      Benin   
35                   Botswana   
36               Burkina Faso   
97               Burkina Faso   
37               Burkina Faso   
39               Burkina Faso   
38               Burkina Faso   
3                     Burundi   
40                    Burundi   
42                   Cameroon   
41                   Cameroon   
88   Central African Republic   
1         Congo (Brazzaville)   
44              Cote D'Ivoire   
98              Cote D'Ivoire   
43              Cote D'Ivoire   
45                        DRC   
2                         DRC   
99                        DRC   
100                       DRC   
46                    Eritrea   
101                  Ethiopia   
23                   Ethiopia   
89                   Ethiopia   
7                    Ethiopia   
..                        ...   
61               Sierra Leone   
92                 Somaliland   
19                 Somaliland   
17                 Somaliland   
18                 Somaliland   
63               South Africa   
64                South Sudan   
20                      Sudan   
21                  Swaziland   
30                   Tanzania   
28                   Tanzania   
65                   Tanzania   
66                   Tanzania   
29                   Tanzania   
78                       Togo   
91                       Togo   
0                      Uganda   
70                     Uganda   
69                     Uganda   
68                     Uganda   
67                     Uganda   
108                    Uganda   
31                     Uganda   
22                     Uganda   
109                    Uganda   
73                     Zambia   
71                     Zambia   
75                   Zimbabwe   
74                   Zimbabwe   
76                   Zimbabwe   

                                          requirements  \
33                                     Nursing diploma   
94                                      Medical school   
34                                             Diploma   
87                                     Nursing diploma   
86                                     Nursing diploma   
85                                     Nursing diploma   
35                                      Nursing degree   
36                                      Nursing degree   
97                                      Medical school   
37                                     Nursing diploma   
39                                     Nursing diploma   
38                                     Nursing diploma   
3                                     Secondary school   
40                                     Nursing diploma   
42                                     Nursing diploma   
41                                      Nursing degree   
88                                     Nursing diploma   
1                                      Nursing diploma   
44                                     Nursing diploma   
98                                      Medical school   
43                                     Nursing diploma   
45                                     Nursing diploma   
2                                                Other   
99                                      Medical school   
100                                     Medical school   
46                                     Nursing diploma   
101                                     Medical school   
23                                     Nursing diploma   
89                                   Bachelor's degree   
7                                     Secondary school   
..                                                 ...   
61                                      Nursing degree   
92   Previously qualified nurse, midwife or pharmacist   
19 

In [111]:
melt = df2.melt(id_vars=['country', 'degree', 'requirements'])
melt

country                                         degree  \
0                     Angola                                        Diploma   
1                     Angola                           physician specialist   
2                      Benin                                        Diploma   
3                      Benin                                Master's degree   
4                      Benin                                Master's degree   
5                      Benin                                Master's degree   
6                   Botswana                                        Diploma   
7               Burkina Faso                                        Diploma   
8               Burkina Faso                           physician specialist   
9               Burkina Faso                                        Diploma   
10              Burkina Faso                                        Diploma   
11              Burkina Faso                                        Diploma   
12                   Burundi                              Bachelor's degree   
13                   Burundi                                        Diploma   
14                  Cameroon                                        Diploma   
15                  Cameroon                                        Diploma   
16  Central African Republic                                Master's degree   
17       Congo (Brazzaville)  Assistant sanitaire en anesthésie-réanimation   
18             Cote D'Ivoire                                        Diploma   
19             Cote D'Ivoire                           physician specialist   
20             Cote D'Ivoire                                        Diploma   
21                       DRC                                        Diploma   
22                       DRC                                          Autre   
23                       DRC                           physician specialist   
24                       DRC                           physician specialist   
25                   Eritrea                                        Diploma   
26                  Ethiopia                           physician specialist   
27                  Ethiopia                                    Certificate   
28                  Ethiopia                                Master's degree   
29                  Ethiopia                              Bachelor's degree   
..                       ...                                            ...   
69              Sierra Leone                                        Diploma   
70                Somaliland                             Nurse anesthetist'   
71                Somaliland                              Bachelor's degree   
72                Somaliland                              Bachelor's degree   
73                Somaliland                              Bachelor's degree   
74              South Africa                                        Diploma   
75               South Sudan                                        Diploma   
76                     Sudan                              Bachelor's degree   
77                 Swaziland                              Bachelor's degree   
78                  Tanzania                                    Certificate   
79                  Tanzania                                    Certificate   
80                  Tanzania                                        Diploma   
81                  Tanzania                                        Diploma   
82                  Tanzania                                    Certificate   
83                      Togo                                   I don't know   
84                      Togo                                Master's degree   
85                    Uganda                               Advanced Diploma   
86                    Uganda                                        Diploma   
87                    Uganda                                        Dip

24.0    32
36.0    19
48.0    17
18.0    15
12.0     9
60.0     2
2.5      1
6.0      1
2.0      1
9.0      1
72.0     1
Name: duration, dtype: int64